test/hotspot/jtreg/runtime/cds/appcds/aotCache/DiagnosticCommandMBeanTest.java

```diff
@@ -0,0 +1,161 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+
+/*
+ * @test
+ * @summary Sanity test for DiagnosticCommand MBean ability to invoke AOT.end_recording
+ * @requires vm.cds.write.archived.java.heap
+ * @library /test/jdk/lib/testlibrary /test/lib
+ *          /test/hotspot/jtreg/runtime/cds/appcds/aotCache/test-classes
+ * @build DiagnosticCommandMBeanTest
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller -jar app.jar DiagnosticCommandMBeanApp
+ * @run driver DiagnosticCommandMBeanTest
+ */
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanException;
+import javax.management.ReflectionException;
+import javax.management.MalformedObjectNameException;
+
+import jdk.test.lib.cds.CDSAppTester;
+import jdk.test.lib.helpers.ClassFileInstaller;
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class DiagnosticCommandMBeanTest {
+    static final String appJar = ClassFileInstaller.getJarPath("app.jar");
+    static final String mainClass = "DiagnosticCommandMBeanApp";
+    public static void main(String[] args) throws Exception {
+        Tester tester = new Tester();
+        tester.runAOTWorkflow();
+    }
+
+    static class Tester extends CDSAppTester {
+        public Tester() {
+            super(mainClass);
+        }
+
+        @Override
+        public String classpath(RunMode runMode) {
+            return appJar;
+        }
+
+        @Override
+        public String[] vmArgs(RunMode runMode) {
+            return new String[] {
+               "-Xlog:cds+class=trace",
+                "--add-modules=jdk.management"
+            };
+        }
+
+        @Override
+        public String[] appCommandLine(RunMode runMode) {
+            return new String[] {
+                mainClass, runMode.name()
+            };
+        }
+
+        @Override
+        public void checkExecution(OutputAnalyzer out, RunMode runMode) {
+            var name = runMode.name();
+            if (runMode.isApplicationExecuted()) {
+                if(runMode == RunMode.TRAINING) {
+                    out.shouldContain("Hello Leyden " + name);
+                    out.shouldContain("AOT.end_recording invoked successfully");
+                    out.shouldContain("Successfully stopped recording");
+                } else if (runMode == RunMode.ASSEMBLY) {
+                    out.shouldNotContain("Hello Leyden ");
+                } else if (runMode == RunMode.PRODUCTION) {
+                    out.shouldContain("Hello Leyden " + name);
+                    out.shouldContain("AOT.end_recording invoked successfully");
+                    out.shouldContain("Failed to stop recording");
+                }
+                out.shouldNotContain("IOException occurred!");
```

Comment:
The 'IOException occurred!' is never printed. While there are a lot of other exception print and not checked.

Also makes sense to check exit status.